GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -956,10 +959,34 @@ private async Task UpdateBranchNameDisplayAsync()
             await TaskScheduler.Default;
 
             var currentBranchName = Module.GetSelectedBranch();
+            if (_branchNameLabelOnClick != null)
+            {
+                remoteNameLabel.Click -= _branchNameLabelOnClick;
+            }
+
+            var currentBranch = Module.GetRefs(false, true).FirstOrDefault(r => r.LocalName == currentBranchName);
+            if (currentBranch == null)
+            {
+                branchNameLabel.Text = currentBranchName;
+                remoteNameLabel.Text = string.Empty;
+                return;
+            }
+
+            var pushTo = string.IsNullOrEmpty(currentBranch.TrackingRemote) || string.IsNullOrEmpty(currentBranch.MergeWith)
+                ? Module.GetRemotes().Any(r => r.Name == "origin") ? $"(default) origin/{currentBranchName}" : _statusBarBranchWithoutRemote.Text
+                : $"{currentBranch.TrackingRemote}/{currentBranch.MergeWith}";
 
             await this.SwitchToMainThreadAsync();
 
-            branchNameLabel.Text = currentBranchName;
+            branchNameLabel.Text = $"{currentBranchName} {char.ConvertFromUtf32(0x25b6)}";
```

Comment:
This symbol looks kind of rough...

What do you all think of the following?
![image](https://user-images.githubusercontent.com/4403806/52783955-a9536f80-30a7-11e9-88e0-22794376a612.png)

Reply:
I will have a look if it looks better but the second proposition of @drewnoakes  was decent.

Reply:
The thin arrow looks nice too.